pkg/agent/client.go

```diff
@@ -534,7 +534,18 @@ func (a *Client) proxyToRemote(connID int64, ctx *connContext) {
 			klog.V(3).InfoS("Exiting proxyToRemote", "connectionID", connID)
 		}
 	}()
-	defer ctx.cleanup()
```

Comment:
where is cleanup getting called now? Or is it not needed?

Reply:
It is still needed. The normal path where I would expect this to be called is in [remoteToProxy](https://github.com/kubernetes-sigs/apiserver-network-proxy/blob/v0.0.32/pkg/agent/client.go#L495). There is also a batch cleanup when the grpc stream ends in [Serve](https://github.com/kubernetes-sigs/apiserver-network-proxy/blob/v0.0.32/pkg/agent/client.go#L351). If the other end of the connection closes it we should attempt to close based on the [close request](https://github.com/kubernetes-sigs/apiserver-network-proxy/blob/v0.0.32/pkg/agent/client.go#L466).

Reply:
Got it. So those other loops, presumably in other go-routines, are sharing the same connContext and will call `cleanup`, so it's not necessary to call cleanup in this method?

Reply:
Yes.

As it stand we spin up at least two goroutines per connContext, one to run remoteToProxy and one to run proxyToRemote. Today they both attempt to do the cleanup. So it should no be necessary for cleanup to be called here. Given this is the consumer side of the channel being cleaned up, it is arguably wrong for the cleanup to be called here.